sklearn/utils/graph.py

```diff
@@ -119,18 +122,27 @@ def graph_laplacian(csgraph, normed=False, return_diag=False):
 
     if sparse.isspmatrix(csgraph):
         return _laplacian_sparse(csgraph, normed=normed,
-                                 return_diag=return_diag)
+                                 return_diag=return_diag, copy=copy)
     else:
         return _laplacian_dense(csgraph, normed=normed,
-                                return_diag=return_diag)
+                                return_diag=return_diag, copy=copy)
 
 
-def _laplacian_sparse(graph, normed=False, return_diag=False):
+def _laplacian_sparse(graph, normed=False, return_diag=False, copy=True):
     n_nodes = graph.shape[0]
-    if not graph.format == 'coo':
-        lap = (-graph).tocoo()
+    if not copy:  # use the same matrix
+        if not graph.format == 'coo':
+            lap = graph.tocoo(copy=False)  # prevent making a new copy
+            np.negative(lap.data, out=lap.data)
+        else:
+            lap = graph
+            np.negative(lap, out=lap)
     else:
-        lap = -graph.copy()
+        if not graph.format == 'coo':
+            lap = (-graph).tocoo()
+        else:
+            lap = -graph.copy()
```

Comment:
-graph will already copy.

I think as long as the earliest supported version of scipy supports it, (-graph).tocoo(copy=False) should work in all cases.